keystore/src/connection/platform/wasm/storage.rs

```diff
@@ -26,9 +26,83 @@ use crate::{
 
 use super::WasmConnection;
 
+type InMemoryDB = HashMap<String, HashMap<Vec<u8>, JsValue>>;
+
 pub enum WasmStorageWrapper {
     Persistent(idb::Database),
-    InMemory(HashMap<String, HashMap<Vec<u8>, JsValue>>),
+    InMemory(Rc<RefCell<InMemoryDB>>),
+}
+
+// The lifetime is to comply with the sqlite implementation.
+pub enum WasmStorageTransaction<'a> {
+    Persistent {
+        tx: idb::Transaction,
+        cipher: &'a aes_gcm::Aes256Gcm,
+    },
+    InMemory {
+        db: Rc<RefCell<InMemoryDB>>,
+        cipher: &'a aes_gcm::Aes256Gcm,
+    },
+}
```

Comment:
The comment reads like we don't really need the lifetime on a field, just on the type itself. 
Would it be better then to just do the following then? 

```rust
pub enum WasmStorageTransaction<'_a> {
    Persistent {
        tx: idb::Transaction,
        cipher: aes_gcm::Aes256Gcm,
    },
    InMemory {
        db: Rc<RefCell<InMemoryDB>>,
        cipher: aes_gcm::Aes256Gcm,
    },
}
```

Reply:
This doesn't compile. Rust doesn't accept specifying generic types or lifetimes that are not used.